0/a-burke/molecule.py

```diff
@@ -0,0 +1,85 @@
+#Author: Ally Burke
+
+
+import sys
+sys.path.append('../../extra-files')
+import copy
+import numpy as np
+from masses import get_charge
+from masses import get_mass
+
+class Molecule(object):
+
+	#Used to initialize an instance of the Molecule class
+
+	def __init__(self, geom_str, units = "Angstroms"):
+		self.units = units				#Specifies units for spatial coordinates
+		self.read(geom_str)				#Reads in the geometry file for the specific molecule	
+		self.natom = self.read(geom_str)		#Specifies the number of atoms in the molecule
+		self.labels = self.read(geom_str)		#Specifies list of atom symbols following the input file	
+		
+		#unsure how to reference parameters without hardcoding
+		self.masses = get_mass("H") 	        	#Reads in masses from input file
+		self.charges = get_charge("H")			#Reads in charges from input file
+
+	#Unsure if read function is necessary for this project
+
+	def read(self, geom_str):
+		
+		#Read in the file containing the geometry of the molecule in question
+	
+        	geom = []
+        	self.label = []
+        	lines = geom_str.strip().splitlines()
+        	natom = int(lines[0])
+        	for line in lines[2:]:
+           		atom, x, y, z = line.split() 
+           		self.label.append(atom)
+           		geom.append([float(x), float(y), float(z)])
+        	self.geom = np.array(geom)
+
+	#Returns a string representing the contents of the Molecule object in .xyz format
+
+	def xyz_string(self):
+		string = "{}\n{}\n".format(self.natom, self.units)
+		lines = geom_str.strip().splitlines()
```

Comment:
Here this assumes we still have access to geom_str, which isn't true because this isn't a global variable (i.e. it's not self.geom_str). You should be able to perform lines 45-49 using the self.labels and self.geom variables instead.